src/overlays/actors/ovl_En_Kendo_Js/z_en_kendo_js.c

```diff
@@ -28,86 +49,720 @@ const ActorInit En_Kendo_Js_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B27188.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B273D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B2740C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B274BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B276C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B276D8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B27760.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B27774.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B2783C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B27880.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B278C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B279AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B279F0.s")
+static ActorAnimationEntry sAnimations[] = {
+    { &object_js_Anim_000C7C, 1.0f, 0.0f, 0.0f, 0, -8.0f }, { &object_js_Anim_000F4C, 1.0f, 0.0f, 0.0f, 0, -8.0f },
+    { &object_js_Anim_00016C, 1.0f, 0.0f, 0.0f, 0, -8.0f }, { &object_js_Anim_0003DC, 1.0f, 0.0f, 0.0f, 2, -8.0f },
+    { &object_js_Anim_000AD4, 1.0f, 0.0f, 0.0f, 0, -8.0f },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B27A90.s")
+static s16 D_80B27CE0[][3] = {
+    { 0x2731, 0x2732, 0x2733 },
+    { 0x2734, 0x2735, 0x2736 },
+    { 0x2737, 0x2738, 0x2739 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/EnKendoJs_Update.s")
+static s16 D_80B27CF4[][3] = {
+    { 0x2710, 0x2712, 0x2714 },
+    { 0x2711, 0x2713, 0x2715 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B27B54.s")
+static s16 D_80B27D00[] = {
+    0x271C, 0x271E, 0x2720, 0x2722, 0x2724, 0x2726, 0x2728,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/func_80B27B8C.s")
+static s16 D_80B27D10[] = {
+    0x271B, 0x271D, 0x271F, 0x2721, 0x2723, 0x2725, 0x2727,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kendo_Js/EnKendoJs_Draw.s")
+void EnKendoJs_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnKendoJs* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
+
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_js_Skel_006990, &object_js_Anim_000F4C, this->jointTable,
+                       this->morphTable, 0xD);
+
+    if ((CURRENT_DAY == 3) && (!(gSaveContext.time < 0xF556) || (gSaveContext.time < CLOCK_TIME(6, 0)))) {
```

Comment:
```suggestion
    if ((CURRENT_DAY == 3) && !((gSaveContext.time <= CLOCK_TIME(23, 0)) && (gSaveContext.time >= CLOCK_TIME(6, 0)))) {
```
Should still match, and looks cleaner. I prefer the ! with the &&, but could be demorgans to just a || if desired.

Reply:
Time off off-by-one and the condition can't be changed like this.

Reply:
Hmm, I've seen Ido be fine with these kind of demorgans before. I tried checking out your branch, and my suggestion seems fine. I still get matching.